lightshow/postprocess/parse.py

```diff
@@ -0,0 +1,361 @@
+
+    Returns
+    -------
+    dict_output
+        A dictionary containing the spectra from different sites. Each spectrum is averaged from all polarization directions in ``path``.
+        Fermi level and total energy are extracted if ``scf.out`` file is provided, with unit eV
+        ``{
+            "Ti": {"0001_1s": {"spectrum": [..., ...] }, "0002_1s": {"spectrum": [..., ...] } }, 
+            "efermi": ...,
+            "total_energy": ...
+           }``
+
+    Raises
+    ------
+    AssertionError
+        If the output path does not contain any file that starts with ``absspct``. 
+    """
+    
+    dict_output = {}
+    
+    path = Path(path)
+    if scf_out_file is None:
+        scf_out_file = path / "scf.out"
+        
+    num_spectra = 0
+    spectra = None
+    for sub_path in path.rglob('absspct*'): 
+        num_spectra += 1
+        element = sub_path.parts[-1].split('.')[0].split('_')[1]
+        absorber = sub_path.parts[-1].split('.')[1].split('_')[0] + '_' + sub_path.parts[-1].split('.')[1].split('_')[1]
+        if element not in dict_output.keys(): 
+            dict_output[element] = {}
+        else: 
+            if absorber not in dict_output[element].keys(): 
+                dict_output[element][absorber] = {
+                    'spectrum': np.loadtxt( sub_path, usecols=(0,2)  ), 
+                    'num_polar': 1
+                }
+            else: 
+                dict_output[element][absorber]['spectrum'][:,1] += np.loadtxt( sub_path, usecols=(2)  )
+                dict_output[element][absorber]['num_polar'] += 1
+
+    # Raise error if there is no ``absspct...`` file in the directory.                     
+    try: 
+        assert num_spectra > 0, "\'absspct...\' file not found in %s"%path 
+    except AssertionError as msg: 
+        print(msg)
+        return
+    
+    for element in dict_output.keys(): 
+        for absorber in dict_output[element].keys():
+            dict_output[element][absorber]['spectrum'][:,1] /= dict_output[element][absorber]['num_polar']
+            del dict_output[element][absorber]['num_polar']
+
+    # Try to extract Fermi energy from ``scf.out`` file. 
+    file_out = scf_out_file
+    try: 
+        with open(file_out, 'r') as f:
+            for line in f.readlines():
+                if 'Fermi energy' in line:
+                    efermi = float(line.split()[4])
+                if '!' in line:
+                    total_energy = float(line.split()[-2]) * 13.605703976
+        try: 
+            dict_output['efermi'] = efermi
+        except NameError as err: 
+            print('Cannot find Fermi energy in', file_out)
+        try: 
+            dict_output['total_energy'] = total_energy
+        except: 
+            print('Cannot find total energy in', file_out) 
+    except OSError: 
+        print(file_out, 'not found. Fermi energy and total energy not extracted.')
+    
+    return dict_output
+
+
+def extract_exciting(path, INFO_out_file=None): 
```

Comment:
@FabiPi3 Can you post some sample code using the excitingtools library, so that @chuntian236 can take a look.

Reply:
Not sure whether its worth. The spectra files are already very nicely parsed. For the INFO.OUT one could use:

```
from excitingtools.exciting_dict_parsers.groundstate_parser import parse_info_out
scl_info = parse_info_out(INFO_out_file)['scl']
last_scl_info = scl_info[list(scl_info)[-1]]
fermi_energy = float(last_scl_info["Fermi energy"])  # still in Hartree
total_energy = float(last_scl_info["Total energy"])
```

Reply:
Dear Fabian, thank you for the comments! I updated the parse.py file today. I decided not to import excitingtools for now, because the parsing task here is easy (at least at this point).